python/ray/dag/compiled_dag_node.py

```diff
@@ -1016,6 +1097,20 @@ def execute(
 
         self._get_or_compile()
 
+        if self._execution_index - self._max_execution_index > self._capacity:
```

Comment:
I guess not directly related to this PR, but `_max_execution_index` is a confusing name, it sounds like it's the max execution index allowed. But actually it is more like the `_last_output_execution_index`?

Reply:
Changed it to `_last_materialized_execution_index`, let me know if you have better names.